crates/cli/src/detect.rs

```diff
@@ -0,0 +1,68 @@
+use duct::cmd;
+use std::io;
+use std::path::Path;
+
+/// Find an executable in the `PATH`.
+pub(crate) fn find_executable(exe_name: impl AsRef<Path>) -> Option<std::path::PathBuf> {
```

Comment:
There's also https://github.com/harryfei/which-rs/, but it seems to mostly do stuff we don't need, so I think this is fine for now.